src/Stripe.net/Services/_base/Service.cs

```diff
@@ -10,34 +10,44 @@ namespace Stripe
     using System.Threading.Tasks;
     using Stripe.Infrastructure;
 
+    /// <summary>Abstract base class for all services.</summary>
+    /// <typeparam name="EntityReturned">
+    /// The type of <see cref="IStripeEntity"/> that this service returns.
+    /// </typeparam>
     public abstract class Service<EntityReturned>
         where EntityReturned : IStripeEntity
     {
         private IStripeClient client;
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Service{EntityReturned}"/> class.
+        /// </summary>
         protected Service()
         {
         }
 
-        protected Service(string apiKey)
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Service{EntityReturned}"/> class with a
+        /// custom <see cref="IStripeClient"/>.
+        /// </summary>
+        /// <param name="client">The client used by the service to send requests.</param>
+        protected Service(IStripeClient client)
```

Comment:
why do we have 2 constructors instead of just one with a default value of null? (mostly for my own education)

Reply:
Good question! I think this dates back to Jayme's original design... It's possible that optional arguments weren't even a thing in C# when he started the library.

I think we could plausibly merge both constructors into one, though the current implementation works and I'm unsure if there are tricky differences between method overloading and optional arguments, so I'd rather leave it as is for now.